pkg/tsdb/elasticsearch/data_query.go

```diff
@@ -23,20 +23,25 @@ const (
 )
 
 type elasticsearchDataQuery struct {
-	client      es.Client
-	dataQueries []backend.DataQuery
-	logger      log.Logger
-	ctx         context.Context
-	tracer      tracing.Tracer
+	client               es.Client
+	dataQueries          []backend.DataQuery
+	logger               log.Logger
+	ctx                  context.Context
+	tracer               tracing.Tracer
+	keepLabelsInResponse bool
 }
 
-var newElasticsearchDataQuery = func(ctx context.Context, client es.Client, dataQuery []backend.DataQuery, logger log.Logger, tracer tracing.Tracer) *elasticsearchDataQuery {
+var newElasticsearchDataQuery = func(ctx context.Context, client es.Client, req *backend.QueryDataRequest, logger log.Logger, tracer tracing.Tracer) *elasticsearchDataQuery {
+	_, fromAlert := req.Headers[headerFromAlert]
+	fromExpression := req.GetHTTPHeader(headerFromExpression) != ""
+
 	return &elasticsearchDataQuery{
-		client:      client,
-		dataQueries: dataQuery,
-		logger:      logger,
-		ctx:         ctx,
-		tracer:      tracer,
+		client:               client,
+		dataQueries:          req.Queries,
+		logger:               logger,
+		ctx:                  ctx,
+		tracer:               tracer,
+		keepLabelsInResponse: fromAlert || fromExpression,
```

Comment:
hm, maybe a small comment why we need that?

Reply:
👍